index.js

```diff
@@ -5,41 +5,103 @@
 
 // This script is modified version of
 // https://github.com/brave/brave-ios/blob/development/Client/Frontend/UserContent/UserScripts/Playlist.js
-(function () {
+medias = (async function () {
   // This will be replaced by native code on demand.
   const siteSpecificDetector = null
 
+  /**
+   * Returns a Promise that resolves with a boolean argument indicating whether
+   * the `src` URL could be a blob pointing at a MediaSource object.
+   * @param {string} src
+   * @returns {Promise}
+   */
+  async function isMediaSourceObjectURL (src) {
+    if (!src || !src.startsWith('blob:')) {
+      return false
+    }
+
+    const controller = new AbortController()
+    const signal = controller.signal
+    const maybeAbortFetch = new Promise(resolve =>
+      setTimeout(() => {
+        resolve(false)
+        controller.abort()
+      }, 500)
+    )
+
+    return Promise.any([
+      fetch(src, { signal }).then(() => false).catch(() => true),
+      maybeAbortFetch
+    ])
+  }
+
+  /**
+   * Returns whether given `url` has https protocol.
+   * @param {string} url
+   * @returns {boolean}
+   */
   function isHttpsScheme (url) {
     if (!url || typeof url !== 'string') {
       return false
     }
 
+    if (url.startsWith('blob:')) {
+      url = url.substring(5)
+      // blob: should be absolute path.
+      return url.startsWith('https://')
```

Comment:
I'm pretty sure we do, but can you confirm that we check that the url is `https` in C++ as well? It's fine to additionally check it here, but we can't 100% trust it, especially if we inject the script into non-isolated context.

Reply:
It's done here 👍

https://github.com/brave/brave-core/pull/21332/commits/b4980d16ce3062d76b6d84ff3c1af218a5178252